### PR TITLE
Log build CGO_ENABLED state

### PR DESCRIPTION
**Pull request by alexshtin**
**What changed?**
Log build `CGO_ENABLED` state.

**Why?**
For better debugging issues which are related to `cgo` it is very useful to know if `cgo` was enabled during build or not.

**How did you test it?**
Run locally.

**Potential risks**
No risks.

**Is hotfix candidate?**
No.

### CODE REVIEW

File: cmd/server/main.go [hunk 1]
```diff
@@ -47,6 +47,10 @@ import (
 	"go.temporal.io/server/temporal"
 )
 
+var (
+	cgoEnabled = false
```
Comment (yiminc) [inline cmd/server/main.go:51]: where is this variable used?

Comment (alexshtin) [reply cmd/server/main.go:51]: Just a few lines bellow.

Comment (yiminc) [reply cmd/server/main.go:51]: I'm not sure I follow. Line 146 uses build.InfoData.CgoEnabled which is either defined in cgo.go or nocgo.go. I don't see where this valuable is used.

Comment (alexshtin) [reply cmd/server/main.go]: Sorry, @yiminc. It is leftover from 1st implementation. Thanks for catching this!

**Review by yiminc — APPROVED** [review]

**Pull request merged**